io/include/pcl/io/impl/pcd_io.hpp

```diff
@@ -341,29 +339,36 @@ pcl::PCDWriter::writeBinaryCompressed (const std::string &file_name,
   }
 
   char* temp_buf = static_cast<char*> (malloc (static_cast<std::size_t> (static_cast<float> (data_size) * 1.5f + 8.0f)));
+  unsigned int compressed_final_size = 0;
+  if (data_size != 0) {
   // Compress the valid data
   unsigned int compressed_size = pcl::lzfCompress (only_valid_data, 
                                                    static_cast<std::uint32_t> (data_size), 
                                                    &temp_buf[8], 
                                                    static_cast<std::uint32_t> (static_cast<float>(data_size) * 1.5f));
-  unsigned int compressed_final_size = 0;
-  // Was the compression successful?
-  if (compressed_size)
-  {
-    char *header = &temp_buf[0];
-    memcpy (&header[0], &compressed_size, sizeof (unsigned int));
-    memcpy (&header[4], &data_size, sizeof (unsigned int));
-    data_size = compressed_size + 8;
-    compressed_final_size = static_cast<std::uint32_t> (data_size) + data_idx;
+    // Was the compression successful?
+    if (compressed_size > 0)
+    {
+      char *header = &temp_buf[0];
+      memcpy (&header[0], &compressed_size, sizeof (unsigned int));
+      memcpy (&header[4], &data_size, sizeof (unsigned int));
+      data_size = compressed_size + 8;
+      compressed_final_size = static_cast<std::uint32_t> (data_size) + data_idx;
+    }
+    else
+    {
+  #ifndef _WIN32
+      io::raw_close (fd);
+  #endif
+      resetLockingPermissions (file_name, file_lock);
+      PCL_WARN("[pcl::PCDWriter::writeBinaryCompressed] Error during compression!\n");
+      return (-1);
+    }
   }
   else
   {
-#ifndef _WIN32
-    io::raw_close (fd);
-#endif
-    resetLockingPermissions (file_name, file_lock);
-    throw pcl::IOException ("[pcl::PCDWriter::writeBinaryCompressed] Error during compression!");
-    return (-1);
+    // empty cloud case
+    compressed_final_size = 8 + data_idx;
```

Comment:
I think it would be safest here to set `data_size` to 8 and to set the first 8 bytes (2x `std::uint32_t`) of `temp_buf` to zero.

Reply:
done